File: baekjoon/10845-큐/제문/10845.kt
package baekjoon

import java.util.*

fun main() = with(System.`in`.bufferedReader()) {
    var caseNum = readLine().toInt()
    val queue = LinkedList<Int>()

    while (caseNum-- > 0) {
        val commands = readLine().split(" ")

        when (commands[0]) {
            "push" -> queue.add(commands[1].toInt())
            "pop" -> println(if (queue.isNotEmpty()) queue.poll() else -1)
            "size" -> println(queue.size)
            "empty" -> println(if (queue.isEmpty()) 1 else 0)
            "front" -> println(if (queue.isNotEmpty()) queue.first else -1)
            "back" -> println(if (queue.isNotEmpty()) queue.last else -1)
        }
    }
}

File: baekjoon/2주차/10845-큐/정우/10845.py
import sys

getN = int(sys.stdin.readline().strip())
arr = []
for _ in range(getN):
    line = sys.stdin.readline().strip().split(' ')
    if line[0] == 'push':
        arr.append(line[1])
    elif line[0] == 'pop':
        if len(arr) > 0:
            print(arr.pop(0))
        else:
            print(-1)
    elif line[0] == 'size':
        print(len(arr))
    elif line[0] == 'empty':
        print(int(not bool(arr)))
    elif line[0] == 'front':
        if len(arr) > 0:
            print(arr[0])
        else:
            print(-1)
    elif line[0] == 'back':
        if len(arr) > 0:
            print(arr[-1])
        else:
            print(-1)